weaviate/templates/apiKeyOctoAISecret.yaml

```diff
@@ -0,0 +1,32 @@
+{{-  $t2vOctoAI := and (index .Values "modules" "text2vec-octoai" "enabled") (index .Values "modules" "text2vec-octoai" "apiKey") -}}
+{{-  $generativeOctoAI := and (index .Values "modules" "generative-octoai" "enabled") (index .Values "modules" "generative-octoai" "apiKey") }}
+{{-  $octoai := or ($t2vOctoAI) ($generativeOctoAI) }}
+{{- if $octoai }}
+
+{{- $apiKeys := list }}
+{{- if $t2vOctoAI }}
+  {{- $apiKeys = append $apiKeys (index .Values "modules" "text2vec-octoai" "apiKey") -}}
+{{- end }}
+{{- if $generativeOctoAI }}
+  {{- $apiKeys = append $apiKeys (index .Values "modules" "generative-octoai" "apiKey") -}}
+{{- end }}
+
+{{- if gt (len ($apiKeys | uniq)) 1 -}}
+  {{- fail "OctoAI modules activated, but their keys differ. Use the same key on all activated OctoAI modules." }}
+{{- end }}
+
+apiVersion: v1
+kind: Secret
+metadata:
+  name: weaviate-octoai
+  labels:
+    app.kubernetes.io/name: weaviate
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+type: Opaque
+data:
+  {{- if (index .Values "modules" "generative-octoai" "apiKey") }}
```

Comment:
nothing blocking, but it could be a good improvement. As you are already filtering out that only one key is being provided in the upper block:
```
{{- if gt (len ($apiKeys | uniq)) 1 -}}
```
Why using that same $apiKeys list variable and pass the item in position 0 as apiKey value? like this it gets way more simple:
```
apiVersion: v1
kind: Secret
metadata:
  name: weaviate-octoai
  labels:
    app.kubernetes.io/name: weaviate
    app.kubernetes.io/managed-by: {{ .Release.Service }}
type: Opaque
data:
    apiKey: {{ index $apiKeys 0 }}
```
As that list will always have 1 item if we get to that point

Reply:
oh, wow, that's a good one!